rust/stackable-cockpit/src/platform/release/spec.rs

```diff
@@ -108,11 +121,85 @@ impl ReleaseSpec {
             .await
     }
 
+    /// Upgrades a release by upgrading individual operators.
+    #[instrument(skip_all, fields(
+        %namespace,
+    ))]
+    pub async fn upgrade_crds(
+        &self,
+        include_products: &[String],
+        exclude_products: &[String],
+        namespace: &str,
+        k8s_client: &Client,
+    ) -> Result<()> {
+        info!("Upgrading CRDs for release");
+
+        include_products.iter().for_each(|product| {
+            Span::current().record("product.included", product);
+        });
+        exclude_products.iter().for_each(|product| {
+            Span::current().record("product.excluded", product);
+        });
+
+        let client = reqwest::Client::new();
+        let operators = self.filter_products(include_products, exclude_products);
+
+        for (product_name, product) in operators {
+            info!("Upgrading CRDs for {product_name}-operator");
+
+            let release_branch = match product.version.pre.as_str() {
+                "dev" => "main".to_string(),
+                _ => {
+                    format!("{}", product.version)
```

Comment:
We should avoid unnamed interpolations.

```suggestion
                    format!("{product_version}", product_version = product.version)
```

But on that, does `product.version` impl `Display` or `Into<String>`? Because then we wouldn't need `format!` here.

Reply:
https://github.com/stackabletech/stackable-cockpit/pull/379/commits/2e6b8a5a878c179850661cd8a08fd63a4d3ff46d